src/main/java/com/example/weatherwaveapi/util/urlbuilder/OpenWeatherUrlBuilder.java

```diff
@@ -0,0 +1,53 @@
+package com.example.weatherwaveapi.util.urlbuilder;
+
+import com.example.weatherwaveapi.config.GeneralSettings;
+import jakarta.ws.rs.core.UriBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OpenWeatherUrlBuilder {
+    private final String apiKey;
+    private final String weatherApiUrl;
+    private final String forecastApiUrl;
+
+    @Autowired
+    public OpenWeatherUrlBuilder(GeneralSettings generalSettings) {
+        this.apiKey = generalSettings.getOpenWeatherMap().getApi().getKey();
+        this.weatherApiUrl = generalSettings.getOpenWeatherMap().getApi().getWeatherUrl();
+        this.forecastApiUrl = generalSettings.getOpenWeatherMap().getApi().getForecastUrl();
+    }
+
+    public String buildWeatherUrl(String str) {
+        return UriBuilder.fromUri(weatherApiUrl)
+                .queryParam("q", str)
+                .queryParam("appid", apiKey)
+                .build()
+                .toString();
+    }
+
+    public String buildForecastUrlForCity(String city) {
+        return UriBuilder.fromUri(forecastApiUrl)
+                .queryParam("q", city)
+                .queryParam("appid", apiKey)
+                .build()
+                .toString();
+    }
```

Comment:
buildWeatherUrl, buildForecastUrlForCity in these methods logic looks similar, also types of parameters would be similar too, so maybe it could be a good idea to create another reusable method and use it?